### PR TITLE
Adds where_filter option using D:P:D:C.

### PR DESCRIPTION
**Pull request by joshrabinowitz**
This required some changes to D:P:D:C, I've pushed those up as https://github.com/bigpresh/Dancer-Plugin-Database/pull/87. 

For discussion @msouth / @bigpresh ; related to https://github.com/bigpresh/Dancer-Plugin-SimpleCRUD/pull/56

### CODE REVIEW

**Comment by joshrabinowitz** [comment]
I think this is good to go.

File: lib/Dancer/Plugin/SimpleCRUD.pm [hunk 1]
```diff
@@ -1003,42 +1016,49 @@ SEARCHFORM
...
-            );
-            $html .= sprintf '&mdash;<a href="%s">Reset search</a></p>',
-                _external_url($args->{dancer_prefix}, $args->{prefix});
+                push(@search_wheres,
+                    "$table_name."
+                    . $dbh->quote_identifier(params->{searchfield})
+                    . ($st eq 'c' ? ' LIKE ' :
+                       $st eq 'nc' ? ' NOT LIKE ' :
+                       $st eq 'ne' ? ' != ' : ' = ')
+                    . $dbh->quote($search_value));
```
Comment (joshrabinowitz) [inline lib/Dancer/Plugin/SimpleCRUD.pm]: @bigpresh What do you think about changing all $dbh->quote() constructs to pushes into @binds? Consistency blah blah.

Comment (bigpresh) [inline lib/Dancer/Plugin/SimpleCRUD.pm]: ISTR that you can't use bind vars when using LIKE / NOT LIKE, but I'm not certain on that right now.

Comment (joshrabinowitz) [inline lib/Dancer/Plugin/SimpleCRUD.pm]: Hm, I'm pretty sure you can do that. Will investigate and report back.

Comment (joshrabinowitz) [inline lib/Dancer/Plugin/SimpleCRUD.pm]: Just tested it. This works:

    my $rows = $dbh->selectall_arrayref( "select * from simulations where end_value like ?", {}, '%0%' );

Comment (joshrabinowitz) [inline lib/Dancer/Plugin/SimpleCRUD.pm]: let's use all binds here, or all $dbh->quote()'s. Let me know which way you prefer.

**Comment by bigpresh** [comment]
Sorry for the late reply.  Yeah, all binds sounds good to me!

Also, I've just released D::P::D::Core 0.15 to CPAN with your refactoring, so D::P::D::SimpleCRUD can declare that as a prereq.

**Comment by joshrabinowitz** [comment]
@bigpresh Converted to use all binds. 
Also declared dependency on D:P:C:Core 0.15 and bumped copyright dates/

**Comment by bigpresh** [comment]
Thanks - you're super-efficient!  Will review, merge & deploy ASAP.

**Comment by joshrabinowitz** [comment]
Sounds good! Thanks @bigpresh

**Pull request merged**